cmds/webboot/webboot.go

```diff
@@ -0,0 +1,157 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"net/url"
+	"os"
+	"path/filepath"
+	"regexp"
+
+	ui "github.com/gizak/termui/v3"
+	"github.com/u-root/webboot/pkg/menu"
+)
+
+const (
+	downloadByLinkLabel = "Download by link"
+	tcURL               = "http://tinycorelinux.net/10.x/x86_64/release/TinyCorePure64-10.1.iso"
+)
+
+// bookmark record the list of iso webboot allow user to download
+var bookmark = map[string]DownloadedDistro{
+	"tinycore": DownloadedDistro{
+		isDefault: false,
+		url:       tcURL,
+		label:     "Download Tinycore v10.1",
+		isoName:   "TinyCorePure64-10.1.iso",
+		group:     "Tinycore",
+	},
+}
+
+// BootMedia contains required information of a bootable iso
+// no matter it is cached or downloaded
+type BootMedia interface {
+	Path() string
+	Name() string
+	// todo: may have Kernel(), Config() etc later
+}
+
+// DownloadedDistro contains information of the iso user want to download
+type DownloadedDistro struct {
+	isDefault bool
+	label     string
+	isoName   string
+	group     string
+	url       string
+	path      string
+}
+
+// CachedDistro contains information of the iso cached in the memory
+type CachedDistro struct {
+	isDefault bool
+	label     string
+	group     string
+	path      string
+	isoName   string
+}
+
+// GoBack is the backward option
```

Comment:
I think that instead of building a back option into the webboot cmd, you'd want to build it into your menu pkg. But maybe lets remove this for now, and wait until we've got the basic functionality working before thinking about the back option.